packages/service/src/utils.ts

```diff
@@ -263,3 +278,274 @@ export function determineNativeContext(
+        timeTaken = endTime - startTime;
+        console.log(\`Time taken to take a visual snapshot of ${id}: \${timeTaken}ms\`);
+    });
+    `
+    return it
+}
+
+/**
+ * Write the test file
+ */
+function writeTestFile(directoryPath: string, fileID: string, log: Logger, testContent: string) {
+    const filePath = join(directoryPath, `${fileID}.test.js`)
+    try {
+        writeFileSync(filePath, testContent)
+        log.info(`Test file created at: ${filePath}`)
+    } catch (err) {
+        console.error(`It seems that the writing the file to '${filePath}' didn't succeed due to the following error: ${err}`)
+        process.exit(1)
+    }
+}
+
+/**
+ * Create the test content
+ */
+function createTestContent ({ clip, clipSelector, stories, storybookUrl }:CreateTestContent):string {
+    return stories.reduce((acc, storyData) => acc + itFunction(clip, clipSelector, storyData, storybookUrl), '')
+}
+
+/**
+ * Create the file data
+ */
+function createFileData (describeTitle: string, testContent: string):string {
+    return `\ndescribe(\`${describeTitle}\`, () => {\n    ${testContent}\n});\n`
+}
+
+/**
+ * Create the test files
+ */
+export function createTestFiles({
+    clip,
+    clipSelector,
+    directoryPath,
+    log,
+    numShards,
+    storiesJson,
+    storybookUrl,
+}: CreateTestFileOptions) {
+    const storiesArray = Object.values(storiesJson)
+        // By default only keep the stories, not the docs
+        .filter((storyData: StorybookData) => storyData?.type === 'story' || !storyData.parameters?.docsOnly)
+    const fileNamePrefix = 'visual-storybook'
+
+    if (numShards === 1){
+        const testContent = createTestContent({ clip, clipSelector, stories: storiesArray, storybookUrl })
+        const fileData = createFileData('All stories', testContent)
+        writeTestFile(directoryPath, `${fileNamePrefix}-1-1`, log, fileData)
+    } else {
+        const totalStories = storiesArray.length
+        const storiesPerShard = Math.ceil(totalStories / numShards)
+
+        for (let shard = 0; shard < numShards; shard++) {
+            const startIndex = shard * storiesPerShard
+            const endIndex = Math.min(startIndex + storiesPerShard, totalStories)
+            const shardStories = storiesArray.slice(startIndex, endIndex)
+            const testContent = createTestContent({ clip, clipSelector, stories: shardStories, storybookUrl })
+            const fileId = `${fileNamePrefix}-${shard + 1}-${numShards}`
+            const describeTitle = `Shard ${shard + 1} of ${numShards}`
+            const fileData = createFileData(describeTitle, testContent)
+
+            writeTestFile(directoryPath, fileId, log, fileData)
+        }
+    }
+}
+
+/**
+ * Create the storybook capabilities based on the specified browsers
+ */
+export function createStorybookCapabilities(capabilities: Capabilities.RemoteCapabilities, log: Logger) {
+    const isHeadless = process.argv.includes('--headless')
```

Comment:
Does it make sense to set `headless` to `true` as default?

Reply:
I'm not fond of headless, IMHO it's a "bad practice" because it's not what the end user uses. Speed in this case is not the holy grail, but that's me

Reply:
Even though I'm not a fan of headless, I've made it the default. I've checked more projects and devs love headless.